cmd/storage-consumer/main.go

```diff
@@ -272,21 +281,28 @@ func newConsumer(ctx context.Context) (*consumer, error) {
 	}
 
 	errCh := make(chan error, 1)
-	s, err := sink.New(ctx, model.DefaultChangeFeedID("storage-consumer"),
-		downstreamURIStr, config.GetDefaultReplicaConfig(), errCh)
+	stdCtx := contextutil.PutChangefeedIDInCtx(ctx,
+		model.DefaultChangeFeedID(defaultChangefeedName))
+	factory, err := factory.New(
+		stdCtx,
+		downstreamURIStr,
+		config.GetDefaultReplicaConfig(),
+		errCh,
+	)
 	if err != nil {
 		log.Error("failed to create sink", zap.Error(err))
 		return nil, err
 	}
 
 	return &consumer{
-		sink:            s,
+		sinkFactory:     factory,
 		replicationCfg:  replicaConfig,
 		codecCfg:        codecConfig,
 		externalStorage: storage,
 		fileExtension:   extension,
 		tableIdxMap:     make(map[dmlPathKey]uint64),
 		tableTsMap:      make(map[model.TableID]uint64),
+		tableSinkMap:    make(map[int64]tablesink.TableSink),
```

Comment:
Whether this should be compatible with `span`?

Reply:
The cloud storage sink feature is not compatible with the span-level replication feature, so using TableID as map key is ok.